### PR TITLE
fix #59: exclude irrelevant files for npm publish

### LINKED ISSUE #59: Shrink node_module size
After installing zmq-prebuilt, the `node_module` directory will be 36MB on my machine. This will be a bit too large for a slow connection for first time downloading. There are a lot of dependencies that might only related for the build process. 

I tried to shrink the package to less than 1.5MB by keeping only `bindings` and `nan` and `zmq-prebuilt/build` (sure including all readme and license files). It works just fine. As I'm not a ZMQ heavy user, I don't know, whether my change will break anything. 

If it is not, I think we can remove irrelevant stuffs in postinstall section.

### PR DESCRIPTION
**Pull request by stanleyxu2005**

### CODE REVIEW

**Comment by stanleyxu2005** [comment]
I dont pretty much understand why we have separated modules into `dependencies` and `devDependencies`, but both will be published in npm registry. I usually explicitly specify the files to be published. In this way I can manage the size of the distribution.

**Comment by rgbkrk** [comment]
`devDependencies` are only installed when you `npm install` within that package's directory - for example, when you clone from git.

`dependencies` are required for those who install from npm: `npm install module-name`.

* [Longer form explanation on Stack Overflow](http://stackoverflow.com/a/22004559/700228)
* [npm - Dev Dependencies](https://docs.npmjs.com/files/package.json#devdependencies)

**Comment by rgbkrk** [comment]
> I usually explicitly specify the files to be published.

The way that we handle this in other packages is [with a `.npmignore`](https://docs.npmjs.com/files/package.json#files). Either approach works.

File: package.json [hunk 1]
```diff
@@ -71,5 +71,14 @@
...
     "Bent Cardan <bent@nothingsatisfies.com> (https://github.com/reqshark)"
+  ],
+  "files": [
+    "HISTORY.md",
+    "LICENSE",
+    "README.md",
+    "index.js",
+    "package.js",
+    "lib/",
+    "build/Release/zmq.node"
```
Comment (rgbkrk) [inline package.json]: If we don't include the `windows` directory, we won't be able to rebuild for Electron.

**Comment by rgbkrk** [comment]
Thank you for putting this together, we definitely need to trim this package down to what is necessary.

File: package.json (hunk 1 above)
Comment (rgbkrk) [inline package.json]: This should not include `build/Release/zmq.node` - this file would only be the one system we publish from (which would mean that, for us, you're going to end up with an OS X node file when you need one for Windows).

**Comment by rgbkrk** [comment]
This also misses `binding.gyp` and many other files that are needed to allow rebuilds.

**Comment by rgbkrk** [comment]
Since we've taken a new approach with the Windows assets and feedback wasn't addressed, I think we can close this. Thanks for the PR @stanleyxu2005.

**Comment by stanleyxu2005** [comment]
Agree. Thanks.

**Pull request closed without merging**